e2e/test/scenarios/filters/reproductions/31340-long-column-name-search-results.cy.spec.js

```diff
@@ -0,0 +1,61 @@
+import { getBrokenUpTextMatcher, popover, restore } from "e2e/support/helpers";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+
+const { PEOPLE_ID } = SAMPLE_DATABASE;
+describe("issue 31340", function () {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.createQuestion(
+      {
+        query: {
+          "source-table": PEOPLE_ID,
+          limit: 2,
+        },
+      },
+      { visitQuestion: true },
+    );
+
+    cy.intercept("GET", "/api/field/*/search/*").as("search");
+  });
+
+  it("should properly display long column names in filter options search results (metabase#31340)", () => {
+    const columnName = "Password";
+
+    cy.findAllByTestId("header-cell").contains(columnName).click();
+
+    popover().within(() => {
+      cy.findByText("Filter by this column").click();
+
+      cy.findByPlaceholderText(`Search by ${columnName}`).type(
+        "nonexistingvalue",
+      );
+
+      cy.wait("@search");
+
+      cy.findByText(getBrokenUpTextMatcher(`No matching ${columnName} found.`))
+        .should("be.visible")
+        .then($container => {
+          cy.findByText(columnName).then($columnTextEl => {
+            // TODO [32181]: this is needed to emulate a column with long name. Have to do it like this until we fix the mentioned issue.
+            $columnTextEl[0].textContent =
+              "Some very very very very long column name that should have a line break";
```

Comment:
This is reproducible with the steps from https://github.com/metabase/metabase/issues/31340, though the description provided there is a bit vague.

The Data Model tab is now called Table Metadata, and you could change the column name there. Also, @kamilmielnik [mentioned this](https://github.com/metabase/metabase/issues/32181#issuecomment-1624859494) in the new issue you filed.
![image](https://github.com/metabase/metabase/assets/1937582/697ac944-3618-4cf7-8898-834f93b64cbb)